runs/views.py

```diff
@@ -0,0 +1,78 @@
+import io
+from typing import TYPE_CHECKING, Sequence
+
+from django.http import HttpRequest
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.exceptions import NotFound
+from rest_framework.parsers import JSONParser
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.viewsets import ViewSet
+
+from freva import settings
+from toil.client import RunWorkflow, ToilClient
+from workflows.models import Workflow
+
+from .models import Run
+from .serializers import CreateRunSerializer, RunSerializer
+
+if TYPE_CHECKING:
+    from rest_framework.permissions import (
+        _PermissionClass,  # type: ignore [reportPrivateUsage]
+    )
+
+
+class RunViewSet(ViewSet):
+    permission_classes: Sequence["_PermissionClass"] = [IsAuthenticated]
+
+    def list(self, request: HttpRequest) -> Response:
+        runs = Run.objects.all()
+        return Response(RunSerializer(runs, many=True).data)
+
+    def create(self, request: HttpRequest) -> Response:
+        toil = ToilClient(settings.TOIL["host"], settings.TOIL["port"])
+        stream = io.BytesIO(request.body)
```

Comment:
That's really strange because it works fine for me. And curiously, I have these functions marked as taking a django `HttpRequest` which all the type checkers are totally fine with. Switching it to rest framework's `Request` means it can now find `data` so I guess it's a property of that and not `HttpRequest`. Though I'm not sure why the type checkers were totally fine. Maybe it can't evaluate the proper types for these.